### PR TITLE
Header

### PR DESCRIPTION
**Pull request by Roman15122022**
Semi-final PR

### CODE REVIEW

File: src/app/services/widgetservice/widget.service.ts [hunk 1]
```diff
@@ -49,4 +50,31 @@ export class WidgetService {
...
+    const screenWidth = window.innerWidth;
+    let slideConfig = {
+      slidesToScroll: 3,
+      slidesToShow: 3,
+      prevArrow: 0,
+      nextArrow: 0,
+    }
+
+    if (screenWidth < 1000 && screenWidth >= 687) {
+      return {
```
Comment (Nikemaks) [inline src/app/services/widgetservice/widget.service.ts]: You can not duplicate property:
```
if (screenWidth < 1000 && screenWidth >= 687) {
slideConfig.slidesToScroll = 2
slideConfig:slidesToShow: 2,
}
```

File: src/app/widget/widget.component.ts [hunk 2]
```diff
@@ -16,13 +19,27 @@ export class WidgetComponent implements OnInit {
...
     new WidgetUiMode({} as WeatherWidget),
+    new WidgetUiMode({} as WeatherWidget),
+    new WidgetUiMode({} as WeatherWidget),
+    new WidgetUiMode({} as WeatherWidget),
   ];
+  @ViewChild('slickModal', { static: true }) slickModal!: SlickCarouselComponent;
+  @ViewChild('removeLast') removeLast!: MatButton;
+  @ViewChild('btnLeft') btnLeft!: ElementRef;
+  @ViewChild('btnRight') btnRight!: ElementRef;
+  @HostListener('window:resize', ['$event'])
```
Comment (Nikemaks) [inline src/app/widget/widget.component.ts]: please add space

File: src/app/widget/widget.component.ts [hunk 3]
```diff
@@ -16,13 +19,27 @@ export class WidgetComponent implements OnInit {
...
+  onResize() {
+    this.slideConfig = this.widgetService.getConfigBySize();
+    console.log(this.slideConfig)
```
Comment (Nikemaks) [inline src/app/widget/widget.component.ts]: remove console

File: src/app/widget/widget.component.ts [hunk 4]
```diff
@@ -80,12 +95,31 @@ export class WidgetComponent implements OnInit {
...
   }
 
-  nextSlide() {
-    /*ToDo*/
+  addWidget(){
+    const newWidget = new WidgetUiMode({} as WeatherWidget);
+    this.weatherWidgets.push(newWidget);
+    if (this.weatherWidgets.length > 3){
+      this.removeLast.color = 'accent';
+      this.btnLeft.nativeElement.classList.remove('display_none');
```
Comment (Nikemaks) [inline src/app/widget/widget.component.ts]: you can toggle this button via ngIf

**Review by Nikemaks — COMMENTED** [review]
Good job, but need to fix small issues.

File: src/app/widget/widget.component.ts (hunk 4 above)
Comment (Roman15122022) [reply src/app/widget/widget.component.ts]: but for adaptive it will be better like this, on my opinion

**Pull request merged**